From 31ea2b8c508eea0e0b63d12a34fee42923fc1cef Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Tue, 4 Jul 2017 15:53:36 +0800
Subject: [PATCH 2324/5242] MLK-16136-2 ARM64: dts: imx8mq: add irqsteer node.

commit  8d3216b6504b42e6b170087952af50208fd827c4 from
https://source.codeaurora.org/external/imx/linux-imx.git

The DCSS interrupts go to GIC via irqsteer module.
So add this device node on imx8mq board.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi |   12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
index f8234f4..00908a8 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
@@ -306,6 +306,18 @@
 		clock-names = "ipg", "per", "osc_per";
 	};
 
+	irqsteer_dcss: irqsteer@32e2d000 {
+		compatible = "nxp,imx-irqsteer";
+		reg = <0x0 0x32e2d000 0x0 0x1000>;
+		interrupts = <GIC_SPI 18 IRQ_TYPE_LEVEL_HIGH>;
+		interrupt-controller;
+		interrupt-parent = <&gpc>;
+		#interrupt-cells = <2>;
+		nxp,irqsteer_chans = <2>;
+		clocks = <&clk IMX8MQ_CLK_DUMMY>;
+		clock-names = "ipg";
+	};
+
 	lcdif: lcdif@30320000 {
 		compatible = "fsl,imx8mq-lcdif", "fsl,imx28-lcdif";
 		reg = <0x0 0x30320000 0x0 0x10000>;
-- 
1.7.9.5

